File: javascript/tests/test_requests.js
/* globals describe, it, before, after */
const assert = require('assert')
const sinon = require('sinon')
const axios = require('axios')
const smoloki = require('../dist/smoloki.js')

describe('push', () => {
  let clock

  before(() => {
    clock = sinon.useFakeTimers(1673798670922)
  })

  after(() => {
    clock.restore()
  })

  it('works', async () => {
    const post = sinon.replace(axios, 'post', sinon.fake())

    await smoloki.push(
      { service: 'web' },
      { level: 'info', event: 'visit', session: 'icfhr9iyu34' },
      'host'
    )

    assert.ok(post.calledOnce)
    assert.equal(post.firstCall.args[0], 'host/loki/api/v1/push')
    assert.deepEqual(post.firstCall.args[1], {
      streams: [
        {
          stream: {
            service: 'web'
          },
          values: [
            [
              '1673798670922000000',
              'level=info event=visit session=icfhr9iyu34'
            ]
          ]
        }
      ]
    }, { headers: {} })
  })
})

File: python/tests/test_requests.py
from unittest.mock import patch, AsyncMock, Mock
import smoloki


def test_push():
    with patch("time.time_ns", return_value=1673798670922295000):
        with patch("aiohttp.ClientSession.post", new=AsyncMock(return_value=Mock())) as post:
            smoloki.push_sync(
                {"service": "web"},
                {"level": "info", "event": "visit", "session": "icfhr9iyu34"},
                base_endpoint="host",
            )

            post.assert_awaited_once_with(
                "host/loki/api/v1/push",
                json={
                    "streams": [
                        {
                            "stream": {
                                "service": "web",
                            },
                            "values": [
                                [
                                    "1673798670922295000",
                                    "level=info event=visit session=icfhr9iyu34",
                                ],
                            ],
                        },
                    ],
                },
                headers={},
            )

            post.return_value.raise_for_status.assert_called_once()
